src/aideck/embeddings/azure_openai_provider.py

```
@@ -0,0 +1,449 @@
+    enable_content_filtering: bool = Field(True, description="Enable Azure content filtering")
+    data_residency_region: Optional[str] = Field(None, description="Data residency region")
+    
+    parallel_requests: int = Field(3, description="Parallel requests (Azure has lower limits)")
+    requests_per_minute: int = Field(240, description="Requests per minute for Azure")
+    tokens_per_minute: int = Field(240000, description="Tokens per minute for Azure")
+    
+    @validator('azure_endpoint')
+    def validate_azure_endpoint(cls, v):
+        """Validate Azure endpoint format."""
+        if v and not v.startswith('https://'):
+            raise ValueError("Azure endpoint must start with https://")
```

Ruff (0.13.1)

Avoid specifying long messages outside the exception class

(TRY003)